packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx

```diff
@@ -92,6 +97,7 @@ export const ArrayInput = (props: ArrayInputProps) => {
     const formGroupName = useFormGroupContext();
     const formGroups = useFormGroups();
     const finalSource = useWrappedSource(source);
+    const parentSourceContext = useSourceContext();
```

Comment:
I find it strange to call both `useWrappedSource` and `useSourceContext` in the same file, as the first one calls the second one.